packages/dds/tree/docs/wip/Container and Application Schema.md

```diff
@@ -0,0 +1,194 @@
+# Container and Application Schema
+
+# Definitions
+
+-   `container data` : data stored in the Fluid container by the shared tree dds.
+-   `container schema` : a set of constraints it is valid to assume the `container data` meets, and that must be maintained when editing (including through conflicts)
```

Comment:
The main distinction here is where its stored (in the container/dds/document with the tree) instead of in the application (its source tree and some runtime data).

I don't like "root schema" since the application root schema.
Some alternative names to replace container vs application:

For "container":

1. document
2. stored/storage
4. recorded/recording
5. persisted/persistent

For application:
1. view
2. projection

I think my new favorite is: stored schema and view schema.

Since this document is mostly not fluid specific (its mostly conceptual/general), I could rename "container data" to "stored data" (which consists of a "stored document" + its "stored schema").

Does that seem like a good idea?

Reply:
IMO document schema matches most closely, based on how I read the proposal. A node (or tree or document) must have a type, and that type is agreed upon by all users, so the type is _intrinsic_ to the data. You cannot have a node without a type, though the type may be 'any'.

Whether the type is stored seems to be an implementation detail because we could have a transient shared tree that is never persisted, but still has the invariants that every node has a type and that type must be respected by all edits.

Reply:
@CraigMacomber - 'intrinsic schema' is now my favorite.

One reason I like it is that it applies equally to both the commonly understood "document" that starts at the root of the tree as well as to the attached augmentations, which are absent from the "document schema", but still have an intrinsic schema that writers must follow.

Reply:
I have renamed them to "stored" and "view". @DLehenbauer Resolve this if you are fine with these names for now.